Sources/Basic/FileSystem.swift

```diff
@@ -155,6 +158,17 @@ public extension FileSystem {
 
 /// Concrete FileSystem implementation which communicates with the local file system.
 private class LocalFileSystem: FileSystem {
+
+    func isExecutableFile(_ path: AbsolutePath) -> Bool {
+        if Basic.exists(path) {
```

Comment:
I would recommend using stat directly -- there is no need to effectively do two stats here, the stat already tells you if the file exists